rules/firestore.json

```diff
@@ -35,50 +36,188 @@ service cloud.firestore {
                    hasField(message, 'postedTime');
         }
         function hasField(data, field) {
-            return data[field] != null;
+            return field in data && data[field] != null;
         }
         function isMessageAuthor(message) {
             return request.auth.uid == message.sender.id
         }
         function usernameOf(uid) {
-            return get(/databases/$(database)/documents/joueurs/$(uid)).data.username
-        }
-        // joueurs will be renamed to users
-        match /joueurs/{userId} {
-            // Anybody can create a user
-            allow create: if true;
-            // Only a user can update its own fields
-            allow update: if isOwner(userId);
-            // Anybody can read the fields of a user
-            allow read: if true;
+            return get(/databases/$(database)/documents/users/$(uid)).data.username
         }
         match /users/{userId} {
-            // Anybody can create a user
-            allow create: if true;
-            // Only a user can update its own fields
-            allow update: if isOwner(userId);
-            // Anybody can read the fields of a user
+            // Only an authenticated account can create its own user, and they cannot directly be verified
+            allow create: if isOwner(userId) && request.resource.data.verified == false
```

Comment:
rah, techniquement ici du coup on authorise la création de doublons (deux comptes avec même pseudo) ?

Reply:
yep mais ça  c'est pas vérifiable à ce niveau, ça demanderait de faire une requête sur toute la db ce qui n'est pas possible

Reply:
Donc, comme ce n'est pas enforcé par la sécurité, c'est authorisé !
ALLEZ ON PLIE BOUTIQUE :D haha

Reply:
tout à fait ! Si y'a un backend un jour il pourra se charger de ce genre de cas aussi

Reply:
Donc ça c'est du "there is no zero% risk" ou plutôt "ce sera mieux quand y aura un backend" ?

Reply:
plutôt « ça sera mieux avec un backend » oui, car c'est effectivement quelque chose d'évitable, mais je vois pas comment sans le backend

Reply:
Tiens, on commencerais pas à faire un gros ticket-dossier pour mettre tout les check qui devraient finir din l'backend lo ?

Reply:
créé!